mslib/plot.py

```diff
@@ -87,84 +93,55 @@ def __init__(self):
         self.params["basemap"].update(self.config["predefined_map_sections"][section]["map"])
         print(self.params)
         self.bbox_units = self.params["bbox"]
-        self.wps = load_from_ftml(filename)
+        self.wpslist = []
+        self.wpslist = load_from_ftml(filename)
+        self.wps = self.wpslist[0]
+        self.wp_model_data = self.wpslist[1]
         self.wp_lats, self.wp_lons, self.wp_locs = [[x[i] for x in self.wps] for i in [0, 1, 3]]
-        self.wp_presss = [mslib.utils.thermolib.flightlevel2pressure(wp[2] * units.hft).to("Pa").m for wp in self.wps]
+        self.wp_press = [mslib.utils.thermolib.flightlevel2pressure(wp[2] * units.hft).to("Pa").m for wp in self.wps]
         self.fig.clear()
         self.ax = self.fig.add_subplot(111, zorder=99)
         self.path = [(wp[0], wp[1], datetime.datetime.now()) for wp in self.wps]
+        self.vertices = [list(a) for a in (zip(self.wp_lons, self.wp_lats))]
         self.lats, self.lons = mslib.utils.coordinate.path_points([_x[0] for _x in self.path], [_x[1] for _x in self.path],
                                                                   numpoints=self.num_interpolation_points + 1,
                                                                   connection="greatcircle"
                                                                   )
 
 
 class TopViewPlotting(Plotting):
-    def __init__(self):
+    def __init__(self, itime=None, vtime=None, no_of_plots=1):
         super(TopViewPlotting, self).__init__()
+        self.itime = itime
+        self.vtime = vtime
+        self.no_of_plots = no_of_plots
         self.myfig = qt.MyTopViewFigure()
         self.myfig.fig.canvas.draw()
         self.line = None
-
-    def TopViewPath(self):
         matplotlib.backends.backend_agg.FigureCanvasAgg(self.myfig.fig)
         self.ax = self.myfig.ax
         self.fig = self.myfig.fig
         self.myfig.init_map(**(self.params["basemap"]))
         self.myfig.set_map()
+        self.plotter = mpath.PathH_GCPlotter(self.myfig.map)
 
-        # plot path
+    def TopViewPath(self):
```

Comment:
The function names do not follow the PEP8 convention. Also adding TopView here seems redundant.

I think it is sensible to have small functions plotting the individual plot components. However calling three functions without any argument in main() seems like a strange design. Why not only one? *In principle* you could put everything in the constructor - or make a simple function doing all that.

What makes sense is reusing the MyFigure class for all plots of one type. But maybe the draw() function should not contain the loop. Instead the loop should IMO go into the main() function and call a draw function with flightpath etc. as  *arguments*.
I also suspect that this does not work properly for multiple flights as, e.g., the label functions must be called after changing the flight.
But maybe I am mistaken, I haven't tested it in practice.